### ISSUE TITLE
[Bug]: Debugger misses WebSocket packets when page loads external URL/script

### ISSUE THREAD

**Issue by adigerber** [electron/electron#44129]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

32.1.2

### What operating system(s) are you using?

Windows

### Operating System Version

Windows 11

### What arch are you using?

x64

### Expected Behavior

The `webContents.debugger` should not "miss" packets, regardless of whatever happens in the page.

### Actual Behavior

The `webContents.debugger` misses packets that are sent by a WebSocket server that load external scripts/UI/iframe (or specifically, reCAPTCHA)

### Testcase Gist URL

https://gist.github.com/adigerber/fd0df6a1f70c133dd46bc26cad4848d5

### Additional Information

Hi,
In the gist I've provided:
* The application creates a WebSocketServer (**note: you need to add the `ws` module to the fiddle**) that sends packets of an ever-increasing counter.
* The application creates a basic window and uses `webContents.debugger` with network enabled to listen to received packets. When a packet is received, it prints its payload. When the connection closes, it prints the total amount of packets that it caught.
* The renderer - a basic web application that connects to the aforementioned WebSocketServer and displays the data it receives. When it receives "10", it renders a reCAPTCHA v2 challenge explicitly (note: the configured site key does not have to be valid for the example). When it receives "15", it closes the connection and does nothing further.

Basically, there's an application that monitors the WebSocket data that is received by the webapp. However, when "10" is received by the webapp it renders the reCAPTCHA challenge (a bogus one but that is irrelevant; it's rendered at the bottom right hand of the window) BUT the debugger does NOT emit a `Network.webSocketFrameReceived` for that "10" message... which is the bug I'm reporting.
At the end of it, the application closes the WebSocket when it receives "15" from the server, at which point:
* The webapp has logged 15 packets that were received
* The WebSocketServer has logged that it has sent 15 packets
* The debugger has logged that it received 14 packets, and "10" is missing.

**Comment by adigerber** [comment]
Reproduced on Ubuntu 22.04 as well.
